[Bug] PAX: build failed in Ubuntu 24.04.03 / gcc/g++ 13.3.0

### Apache Cloudberry version

main

### What happened

When running 

```
build pax, USE_MANIFEST_API=OFF USE_PAX_CATALOG=ON
-- The C compiler identification is GNU 13.3.0
-- The CXX compiler identification is GNU 13.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- pg_config.h => /home/gpadmin/cloudberry/contrib/pax_storage/../../src/include/pg_config.h
-- Looking for USE_ASSERT_CHECKING
-- Looking for USE_ASSERT_CHECKING - found
-- Looking for USE_LZ4
-- Looking for USE_LZ4 - found
-- PAX BUILD TYPE: Debug
-- Looking for USE_VECTORIZATION
-- Looking for USE_VECTORIZATION - not found
-- Build pax without vectorization support
-- Found Protobuf: /usr/lib/aarch64-linux-gnu/libprotobuf.so (found version "3.21.12")
-- Found Protobuf: /usr/lib/aarch64-linux-gnu/libprotobuf.so (found suitable version "3.21.12", minimum required is "3.5.0")
-- Checking for module 'libzstd'
--   Found libzstd, version 1.5.5
-- tabulate version: v1.5.0
-- Using C++11
-- Found Threads: TRUE
-- Configuring done (0.5s)
-- Generating done (0.0s)
-- Build files have been written to: /home/gpadmin/cloudberry/contrib/pax_storage/build
cd build && make -j8
make[2]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[2]: warning: -j8 forced in submake: resetting jobserver mode.
make[3]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[  0%] Building C object src/cpp/CMakeFiles/generate_sql_script_program.dir/__/__/tools/gen_sql.c.o
[  0%] Building CXX object src/cpp/contrib/googletest/googletest/CMakeFiles/gtest.dir/src/gtest-all.cc.o
[  0%] Built target generate_protobuf
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[  1%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/proto/micro_partition_stats.pb.cc.o
[  1%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/proto/pax.pb.cc.o
[  1%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_column.cc.o
[  1%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/proto/orc_proto.pb.cc.o
[  2%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_column_traits.cc.o
[  2%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_compress.cc.o
[  3%] Linking C executable generate_sql_script_program
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[  3%] Built target generate_sql_script_program
[  4%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_columns.cc.o
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[  4%] dynamically generate sql script file
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[  4%] Built target create_sql_script
[  4%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_encoding_utils.cc.o
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[  5%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/proto/micro_partition_stats.pb.cc.o
[  5%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/proto/pax.pb.cc.o
[  5%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_encoding_non_fixed_column.cc.o
[  5%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/proto/orc_proto.pb.cc.o
[  6%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_column_traits.cc.o
[  6%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_column.cc.o
[  6%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_compress.cc.o
[  7%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_encoding_column.cc.o
[  8%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_columns.cc.o
[  8%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_encoding_utils.cc.o
[  8%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_dict_encoding.cc.o
[  9%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_decoding.cc.o
[  9%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_encoding_non_fixed_column.cc.o
[ 10%] Linking CXX static library ../../../../../lib/libgtest.a
[ 10%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_encoding.cc.o
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[ 10%] Built target gtest
[ 11%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_encoding_column.cc.o
[ 11%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_rlev2_decoding.cc.o
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[ 11%] Building CXX object src/cpp/contrib/googletest/googlemock/CMakeFiles/gmock.dir/src/gmock-all.cc.o
[ 12%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_rlev2_encoding.cc.o
[ 12%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_dict_encoding.cc.o
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[ 12%] Building CXX object src/cpp/contrib/googletest/googletest/CMakeFiles/gtest_main.dir/src/gtest_main.cc.o
[ 12%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_decoding.cc.o
[ 13%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_encoding.cc.o
[ 13%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_vec_column.cc.o
[ 13%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_rlev2_decoding.cc.o
[ 13%] Linking CXX static library ../../../../../lib/libgtest_main.a
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[ 13%] Built target gtest_main
[ 13%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_vec_bitpacked_column.cc.o
[ 13%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_rlev2_encoding.cc.o
[ 14%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_vec_bpchar_column.cc.o
[ 14%] Linking CXX static library ../../../../../lib/libgmock.a
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[ 14%] Built target gmock
[ 15%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_vec_bitpacked_column.cc.o
[ 15%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_vec_encoding_column.cc.o
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[ 16%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/proto/micro_partition_stats.pb.cc.o
[ 16%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_vec_numeric_column.cc.o
[ 16%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/proto/pax.pb.cc.o
[ 17%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/oper/pax_oper_udf.cc.o
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[ 18%] Building CXX object src/cpp/contrib/googletest/googlemock/CMakeFiles/gmock_main.dir/src/gmock_main.cc.o
[ 18%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/filter/pax_filter.cc.o
[ 18%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_vec_bpchar_column.cc.o
[ 18%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_vec_column.cc.o
[ 18%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/proto/orc_proto.pb.cc.o
[ 19%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_vec_encoding_column.cc.o
[ 20%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_column_traits.cc.o
[ 20%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_vec_numeric_column.cc.o
[ 20%] Linking CXX static library ../../../../../lib/libgmock_main.a
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[ 20%] Built target gmock_main
[ 20%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/filter/pax_row_filter.cc.o
[ 20%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_column.cc.o
[ 21%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/filter/pax_sparse_filter.cc.o
[ 21%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/oper/pax_oper_udf.cc.o
[ 21%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_compress.cc.o
[ 22%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_columns.cc.o
[ 22%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/filter/pax_sparse_pg_path.cc.o
[ 23%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/filter/pax_filter.cc.o
[ 23%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_encoding_utils.cc.o
[ 23%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/filter/pax_row_filter.cc.o
[ 23%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/filter/pax_sparse_vec_path.cc.o
[ 23%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/filter/pax_sparse_filter.cc.o
[ 24%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/oper/pax_oper.cc.o
[ 24%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_encoding_non_fixed_column.cc.o
[ 24%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/oper/pax_stats.cc.o
[ 25%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/filter/pax_sparse_pg_path.cc.o
[ 26%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_encoding_column.cc.o
[ 26%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/file_system.cc.o
[ 26%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_dict_encoding.cc.o
[ 26%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/filter/pax_sparse_vec_path.cc.o
[ 26%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_decoding.cc.o
[ 27%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/local_file_system.cc.o
[ 28%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_encoding.cc.o
[ 28%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/oper/pax_oper.cc.o
[ 28%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/micro_partition.cc.o
[ 28%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_rlev2_decoding.cc.o
[ 28%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/micro_partition_file_factory.cc.o
[ 28%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_rlev2_encoding.cc.o
[ 29%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/oper/pax_stats.cc.o
[ 30%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/micro_partition_metadata.cc.o
[ 31%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_vec_bitpacked_column.cc.o
[ 31%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/micro_partition_row_filter_reader.cc.o
[ 31%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/file_system.cc.o
[ 31%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_vec_bpchar_column.cc.o
[ 31%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/local_file_system.cc.o
[ 31%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_vec_column.cc.o
[ 32%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/micro_partition.cc.o
[ 32%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/micro_partition_file_factory.cc.o
[ 33%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_vec_encoding_column.cc.o
[ 33%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/micro_partition_stats.cc.o
[ 33%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_vec_numeric_column.cc.o
[ 33%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/micro_partition_metadata.cc.o
[ 34%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/micro_partition_stats_updater.cc.o
[ 34%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/oper/pax_oper_udf.cc.o
[ 34%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/micro_partition_udf.cc.o
[ 35%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/micro_partition_row_filter_reader.cc.o
[ 35%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/orc/orc_dump_reader.cpp.o
[ 36%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/filter/pax_filter.cc.o
[ 37%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/orc/orc_format_reader.cc.o
[ 37%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/micro_partition_stats.cc.o
[ 37%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/micro_partition_stats_updater.cc.o
[ 37%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/filter/pax_row_filter.cc.o
[ 37%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/orc/orc_group.cc.o
[ 38%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/micro_partition_udf.cc.o
[ 38%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/orc/orc_vec_group.cc.o
[ 38%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/filter/pax_sparse_filter.cc.o
[ 39%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/orc/orc_reader.cc.o
[ 40%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/filter/pax_sparse_pg_path.cc.o
[ 40%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/orc/orc_dump_reader.cpp.o
[ 40%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/orc/orc_format_reader.cc.o
[ 40%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/filter/pax_sparse_vec_path.cc.o
[ 40%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/orc/orc_type.cc.o
[ 41%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/orc/orc_group.cc.o
[ 41%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/orc/orc_writer.cc.o
[ 41%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/oper/pax_oper.cc.o
[ 41%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/orc/orc_vec_group.cc.o
[ 42%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/pax_buffer.cc.o
[ 42%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/orc/orc_reader.cc.o
[ 43%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/oper/pax_stats.cc.o
[ 43%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/proto/protobuf_stream.cc.o
[ 44%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/orc/orc_type.cc.o
[ 44%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/toast/pax_toast.cc.o
[ 44%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/orc/orc_writer.cc.o
[ 45%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/wal/pax_wal.cc.o
[ 45%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/file_system.cc.o
[ 45%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/pax_buffer.cc.o
[ 45%] Building C object src/cpp/CMakeFiles/paxformat.dir/storage/wal/paxc_desc.c.o
[ 45%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/local_file_system.cc.o
[ 46%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/pax_itemptr.cc.o
[ 47%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/micro_partition.cc.o
[ 47%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/wal/paxc_wal.cc.o
[ 47%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/proto/protobuf_stream.cc.o
[ 48%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/clustering/clustering.cc.o
[ 48%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/micro_partition_file_factory.cc.o
[ 49%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/pax.cc.o
[ 49%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/micro_partition_metadata.cc.o
[ 49%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/clustering/sorter_tuple.cc.o
[ 49%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/paxc_smgr.cc.o
[ 49%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/clustering/sorter_index.cc.o
[ 50%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/micro_partition_row_filter_reader.cc.o
[ 50%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/toast/pax_toast.cc.o
[ 51%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/clustering/zorder_clustering.cc.o
[ 52%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/strategy.cc.o
[ 52%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/micro_partition_stats.cc.o
[ 52%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/clustering/index_clustering.cc.o
[ 52%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/wal/pax_wal.cc.o
[ 52%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/micro_partition_stats_updater.cc.o
[ 52%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/clustering/lexical_clustering.cc.o
[ 53%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/micro_partition_udf.cc.o
[ 54%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/clustering/zorder_utils.cc.o
[ 54%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/orc/orc_dump_reader.cpp.o
[ 54%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/exceptions/CException.cc.o
[ 54%] Building C object src/cpp/CMakeFiles/pax.dir/storage/wal/paxc_desc.c.o
[ 55%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/wal/paxc_wal.cc.o
[ 55%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/orc/orc_format_reader.cc.o
[ 55%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/micro_partition_iterator.cc.o
[ 56%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/orc/orc_group.cc.o
[ 56%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/comm/bitmap.cc.o
[ 56%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/orc/orc_vec_group.cc.o
[ 56%] Building CXX object src/cpp/CMakeFiles/pax.dir/clustering/clustering.cc.o
[ 57%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/comm/bloomfilter.cc.o
[ 58%] Building CXX object src/cpp/CMakeFiles/pax.dir/clustering/sorter_tuple.cc.o
[ 58%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/orc/orc_reader.cc.o
/home/gpadmin/cloudberry/contrib/pax_storage/src/cpp/storage/micro_partition_iterator.cc: In member function ‘virtual pax::MicroPartitionMetadata pax::internal::MicroPartitionInfoIterator::Next()’:
/home/gpadmin/cloudberry/contrib/pax_storage/src/cpp/storage/micro_partition_iterator.cc:167:19: error: moving a temporary object prevents copy elision [-Werror=pessimizing-move]
  167 |   return std::move(ToValue(tuple));
      |          ~~~~~~~~~^~~~~~~~~~~~~~~~
/home/gpadmin/cloudberry/contrib/pax_storage/src/cpp/storage/micro_partition_iterator.cc:167:19: note: remove ‘std::move’ call
/home/gpadmin/cloudberry/contrib/pax_storage/src/cpp/storage/micro_partition_iterator.cc: In member function ‘virtual pax::MicroPartitionMetadata pax::internal::MicroPartitionInfoParallelIterator::Next()’:
/home/gpadmin/cloudberry/contrib/pax_storage/src/cpp/storage/micro_partition_iterator.cc:378:19: error: moving a temporary object prevents copy elision [-Werror=pessimizing-move]
  378 |   return std::move(ToValue(tuple));
      |          ~~~~~~~~~^~~~~~~~~~~~~~~~
/home/gpadmin/cloudberry/contrib/pax_storage/src/cpp/storage/micro_partition_iterator.cc:378:19: note: remove ‘std::move’ call
cc1plus: all warnings being treated as errors
make[4]: *** [src/cpp/CMakeFiles/pax.dir/build.make:832: src/cpp/CMakeFiles/pax.dir/storage/micro_partition_iterator.cc.o] Error 1
make[4]: *** Waiting for unfinished jobs....
[ 58%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/comm/byte_buffer.cc.o
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[3]: *** [CMakeFiles/Makefile2:310: src/cpp/CMakeFiles/pax.dir/all] Error 2
make[3]: *** Waiting for unfinished jobs....
[ 59%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/orc/orc_type.cc.o
[ 59%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/comm/guc.cc.o
[ 59%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/orc/orc_writer.cc.o
[ 59%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/pax_buffer.cc.o
[ 60%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/comm/paxc_wrappers.cc.o
[ 61%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/pax_itemptr.cc.o
[ 61%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/comm/pax_memory.cc.o
[ 61%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/comm/pax_resource.cc.o
[ 62%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/comm/cbdb_wrappers.cc.o
[ 62%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/proto/protobuf_stream.cc.o
[ 62%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/comm/vec_numeric.cc.o
[ 62%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/vec/arrow_wrapper.cc.o
[ 62%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/pax.cc.o
[ 63%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/paxc_smgr.cc.o
[ 63%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/toast/pax_toast.cc.o
[ 63%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/strategy.cc.o
[ 64%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/vec/pax_porc_adpater.cc.o
[ 65%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/wal/pax_wal.cc.o
[ 65%] Building C object src/cpp/CMakeFiles/test_main.dir/storage/wal/paxc_desc.c.o
[ 65%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/vec/pax_porc_vec_adpater.cc.o
[ 66%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/wal/paxc_wal.cc.o
[ 67%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/vec/pax_vec_adapter.cc.o
[ 67%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/micro_partition_iterator.cc.o
[ 67%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/vec/pax_vec_comm.cc.o
[ 67%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/vec/pax_vec_reader.cc.o
[ 67%] Building CXX object src/cpp/CMakeFiles/test_main.dir/clustering/clustering.cc.o
[ 68%] Building CXX object src/cpp/CMakeFiles/test_main.dir/clustering/sorter_tuple.cc.o
[ 68%] Building CXX object src/cpp/CMakeFiles/test_main.dir/clustering/sorter_index.cc.o
[ 68%] Building CXX object src/cpp/CMakeFiles/test_main.dir/clustering/zorder_clustering.cc.o
[ 69%] Building CXX object src/cpp/CMakeFiles/test_main.dir/clustering/index_clustering.cc.o
[ 69%] Building CXX object src/cpp/CMakeFiles/test_main.dir/clustering/lexical_clustering.cc.o
/home/gpadmin/cloudberry/contrib/pax_storage/src/cpp/storage/micro_partition_iterator.cc: In member function ‘virtual pax::MicroPartitionMetadata pax::internal::MicroPartitionInfoIterator::Next()’:
/home/gpadmin/cloudberry/contrib/pax_storage/src/cpp/storage/micro_partition_iterator.cc:167:19: error: moving a temporary object prevents copy elision [-Werror=pessimizing-move]
  167 |   return std::move(ToValue(tuple));
      |          ~~~~~~~~~^~~~~~~~~~~~~~~~
/home/gpadmin/cloudberry/contrib/pax_storage/src/cpp/storage/micro_partition_iterator.cc:167:19: note: remove ‘std::move’ call
/home/gpadmin/cloudberry/contrib/pax_storage/src/cpp/storage/micro_partition_iterator.cc: In member function ‘virtual pax::MicroPartitionMetadata pax::internal::MicroPartitionInfoParallelIterator::Next()’:
/home/gpadmin/cloudberry/contrib/pax_storage/src/cpp/storage/micro_partition_iterator.cc:378:19: error: moving a temporary object prevents copy elision [-Werror=pessimizing-move]
  378 |   return std::move(ToValue(tuple));
      |          ~~~~~~~~~^~~~~~~~~~~~~~~~
/home/gpadmin/cloudberry/contrib/pax_storage/src/cpp/storage/micro_partition_iterator.cc:378:19: note: remove ‘std::move’ call
[ 69%] Building CXX object src/cpp/CMakeFiles/test_main.dir/clustering/pax_clustering_reader.cc.o
cc1plus: all warnings being treated as errors
make[4]: *** [src/cpp/CMakeFiles/test_main.dir/build.make:832: src/cpp/CMakeFiles/test_main.dir/storage/micro_partition_iterator.cc.o] Error 1
make[4]: *** Waiting for unfinished jobs....
[ 70%] Linking CXX shared library libpaxformat.so
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[ 70%] Built target paxformat
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[3]: *** [CMakeFiles/Makefile2:339: src/cpp/CMakeFiles/test_main.dir/all] Error 2
make[3]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[2]: *** [Makefile:136: all] Error 2
make[2]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[1]: *** [Makefile:69: build] Error 2
make[1]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage'
make: *** [GNUmakefile:32: all] Error 2
make: Leaving directory '/home/gpadmin/cloudberry'
```

```
gpadmin@ubuntu:~/cloudberry$ g++ --version
g++ (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
Copyright (C) 2023 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

### What you think should happen instead

_No response_

### How to reproduce

https://cloudberry.apache.org/docs/deployment/quick-build

Run `make -j$(nproc) -C ~/cloudberry`, and the errors will return.

### Operating System

Ubuntu 24.04.03, gcc/g++ 13.3.0

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/apache/cloudberry/blob/main/CODE_OF_CONDUCT.md).

--- a/contrib/pax_storage/CMakeLists.txt
+++ b/contrib/pax_storage/CMakeLists.txt
@@ -21,7 +21,7 @@ set(CMAKE_CXX_STANDARD 17)
 set(TOP_DIR ${PROJECT_SOURCE_DIR}/../..)
 set(CBDB_INCLUDE_DIR ${TOP_DIR}/src/include)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Werror -Wno-unused-function -Wno-error=ignored-qualifiers -Wuninitialized -Winit-self -Wstrict-aliasing -Wno-missing-field-initializers -Wno-unused-parameter -Wno-clobbered -Wno-sized-deallocation -g")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Werror -Werror=pessimizing-move  -Wno-unused-function -Wno-error=ignored-qualifiers -Wno-error=array-bounds  -Wuninitialized -Winit-self -Wstrict-aliasing -Wno-missing-field-initializers -Wno-unused-parameter -Wno-clobbered -Wno-sized-deallocation -g")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wno-unused-parameter -Wno-parameter-name")
 
 option(USE_MANIFEST_API "Use manifest API" OFF)

--- a/contrib/pax_storage/src/cpp/storage/micro_partition_iterator.cc
+++ b/contrib/pax_storage/src/cpp/storage/micro_partition_iterator.cc
@@ -164,7 +164,7 @@ MicroPartitionMetadata MicroPartitionInfoIterator::Next() {
   Assert(tuple);
 
   tuple_ = nullptr;
-  return std::move(ToValue(tuple));
+  return ToValue(tuple);
 }
 
 void MicroPartitionInfoIterator::Rewind() {
@@ -375,7 +375,7 @@ MicroPartitionMetadata MicroPartitionInfoParallelIterator::Next() {
   Assert(tuple);
 
   tuple_ = nullptr;
-  return std::move(ToValue(tuple));
+  return ToValue(tuple);
 }
 
 void MicroPartitionInfoParallelIterator::Rewind() {

--- a/contrib/pax_storage/src/cpp/storage/micro_partition_stats.cc
+++ b/contrib/pax_storage/src/cpp/storage/micro_partition_stats.cc
@@ -173,7 +173,6 @@ static bool PrepareStatisticsInfoCombine(
 
     auto attr = TupleDescAttr(desc, i);
     auto collation = attr->attcollation;
-    FmgrInfo finfo;
     bool get_pg_oper_succ = false;
 
     funcs.emplace_back(std::make_pair(nullptr, nullptr));
@@ -207,7 +206,6 @@ static bool PrepareStatisticsInfoCombine(
 
       GetStrategyProcinfo(attr->atttypid, attr->atttypid, funcs[i]);
       if (allow_fallback_to_pg) {
-        finfos[i] = {finfo, finfo};
         get_pg_oper_succ =
             GetStrategyProcinfo(attr->atttypid, attr->atttypid, finfos[i]);
       }

--- a/contrib/pax_storage/src/cpp/storage/toast/pax_toast.cc
+++ b/contrib/pax_storage/src/cpp/storage/toast/pax_toast.cc
@@ -331,7 +331,7 @@ static std::pair<Datum, std::shared_ptr<MemoryObject>> pax_make_external_toast(
 std::pair<Datum, std::shared_ptr<MemoryObject>> pax_make_toast(
     Datum d, char storage_type) {
   std::shared_ptr<MemoryObject> mobj;
-  Datum result;
+  Datum result = d;
 
   if (!pax_enable_toast) {
     return {d, nullptr};

